Week3/homework/mongodb/index.js

```diff
@@ -25,27 +48,50 @@ async function findEpisodesExercises(client) {
    */
 
   // Find the title of episode 2 in season 2 [Should be: WINTER SUN]
-
-  console.log(
-    `The title of episode 2 in season 2 is ${"TODO: fill in variable here"}`
-  );
+  const episode2Season2 = await client
+    .db("databaseWeek3")
+    .collection("bob_ross_episodes")
+    .findOne({ episode: "S02E02" });
+  console.log(`The title of episode 2 in season 2 is ${episode2Season2.title}`);
 
   // Find the season and episode number of the episode called "BLACK RIVER" [Should be: S02E06]
 
+  const blackRiverEpisode = await client
+    .db("databaseWeek3")
+    .collection("bob_ross_episodes")
+    .findOne({ title: "BLACK RIVER" });
   console.log(
-    `The season and episode number of the "BLACK RIVER" episode is ${"TODO: fill in variable here"}`
+    `The season and episode number of the "BLACK RIVER" episode is ${blackRiverEpisode}`
   );
 
   // Find all of the episode titles where Bob Ross painted a CLIFF [Should be: NIGHT LIGHT, EVENING SEASCAPE, SURF'S UP, CLIFFSIDE, BY THE SEA, DEEP WILDERNESS HOME, CRIMSON TIDE, GRACEFUL WATERFALL]
+  const cliffEpisodes = await client
+    .db("databaseWeek3")
+    .collection("bob_ross_episodes")
+    .find({ elements: { $in: ["CLIFF"] } })
```

Comment:
Would this work as well?

```suggestion
    .find({ elements: "CLIFF" })
```

Reply:
Yes, both code blocks should return the same result since they both search for the string "CLIFF" in the "elements" field of the documents in the "bob_ross_episodes" collection.

Reply:
exactly, but which in your view is more practical? and which might raise concerns when we allow the user to implicitly have more than he needs?

Reply:
@JouharBirakdar223787 

The concern arises when we allow the user to implicitly have more than they need. If we allow the user to input an array of elements instead of a single value, the first code snippet with $in operator provides a safer option. It ensures that all the user-specified elements are matched correctly. In the second code snippet, if the user provides an array of elements, it will not produce the desired results since it expects a single string value.

Therefore, if we anticipate scenarios where the user may provide multiple elements or if we want to future-proof our code, using the $in operator (first code snippet) is generally more practical and safer. However, if we are certain about the input format and need a simple equality match, the second code snippet may be more suitable.

Reply:
🥇